### PR TITLE
stargazer: init at 1.0.5

### PR DESCRIPTION
**Pull request by gaykitty**
###### Description of changes

Add the Gemini server [stargazer](https://git.sr.ht/~zethra/stargazer/)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5365,6 +5365,11 @@
     githubId = 2430469;
     name = "Gavin Rogers";
   };
+  gaykitty = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:5368]: Put this patch in a separate commit.

File: pkgs/servers/gemini/stargazer/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ lib, fetchFromSourcehut, rustPlatform, installShellFiles, scdoc, ... }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "stargazer";
+  version = "1.0.5";
+
+  src = fetchFromSourcehut {
+    owner = "~zethra";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/servers/gemini/stargazer/default.nix:9]: avoid parameterize pname.

File: pkgs/servers/gemini/stargazer/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    rev = version;
+    sha256 = "sha256-n88X3RJD7PqOcVRK/bp/gMNLVrbwnJ2iwi2rCpsfp+o=";
```
Comment (AndersonTorres) [inline pkgs/servers/gemini/stargazer/default.nix:11]:
```suggestion
    hash = "sha256-n88X3RJD7PqOcVRK/bp/gMNLVrbwnJ2iwi2rCpsfp+o=";
```

File: pkgs/servers/gemini/stargazer/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  cargoSha256 = "sha256-Yqh3AQIOahKz2mLeVNm58Yr6vhjU4aQwN62y3Z5/EJc=";
```
Comment (AndersonTorres) [inline pkgs/servers/gemini/stargazer/default.nix:14]:
```suggestion
  cargoHash = "sha256-Yqh3AQIOahKz2mLeVNm58Yr6vhjU4aQwN62y3Z5/EJc=";
```

File: pkgs/servers/gemini/stargazer/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    installManPage stargazer.ini.5
+    installShellCompletion completions/stargazer.{bash,zsh,fish}
+  '';
+
+  meta = with lib; {
+    description = "A fast and easy to use Gemini server";
+    homepage = "https://sr.ht/~zethra/stargazer/";
+    license = licenses.agpl3Plus;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ gaykitty ];
```
Comment (AndersonTorres) [inline pkgs/servers/gemini/stargazer/default.nix:33]:
```suggestion
    maintainers = with maintainers; [ gaykitty ];
    platforms = platforms.linux;
```

File: pkgs/servers/gemini/stargazer/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
+{ lib, fetchFromSourcehut, rustPlatform, installShellFiles, scdoc, ... }:
```
Comment (AndersonTorres) [inline pkgs/servers/gemini/stargazer/default.nix:1]:
```suggestion
{ lib
, fetchFromSourcehut
, rustPlatform
, installShellFiles
, scdoc
}:
```

We do not use ellipsis here.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/servers/gemini/stargazer/default.nix (hunk 5 above)
Comment (AndersonTorres) [reply pkgs/servers/gemini/stargazer/default.nix:33]: Does it have a changelog page?

Comment (gaykitty) [reply pkgs/servers/gemini/stargazer/default.nix:33]: Does this count? https://git.sr.ht/~zethra/stargazer/refs

File: pkgs/servers/gemini/stargazer/default.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, fetchFromSourcehut
+, rustPlatform
+, installShellFiles
+, scdoc
+, ...
```
Comment (AndersonTorres) [inline pkgs/servers/gemini/stargazer/default.nix:6]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Thanks, and welcome to Nixpkgs!

Please study the possibility of adding a module to NixOS subsystem.

**Comment by gaykitty** [comment]
> Thanks, and welcome to Nixpkgs!
> 
> Please study the possibility of adding a module to NixOS subsystem.

Thx! I was planning on looking into that next.
